bdb/file.c

```diff
@@ -2170,29 +2170,41 @@ static void bdb_appsock(netinfo_type *netinfo, SBUF2 *sb)
         (bdb_state->callback->appsock_rtn)(bdb_state, sb);
 }
 
+extern int gbl_pstack_self;
 void pstack_self(void)
 {
-    char buf[256];
+    if (!gbl_pstack_self)
+        return;
+
+    char cmd[256];
+    char output[20] = "/tmp/pstack.XXXXXX";
```

Comment:
Add pid in output name?

Reply:
We can - mkstemp is guaranteed to make the filename unique - just need temporary space.